### PR TITLE
add spankbd, a 3x5+3 split keyboard

### PR DESCRIPTION
**Pull request by idank**
## Description

Span is a 3x5+3 split keyboard, with 2u, 1.5u, 1u thumb clusters.

The keyboard layout Span uses is derived from GergoPlex, a keyboard that was originally available on gboards.ca
(now defunct).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/idank/spankbd/keymaps/default/config.h
Comment (zvecr) [inline keyboards/idank/spankbd/keymaps/default/config.h:1]: This file should be removed.

File: keyboards/idank/spankbd/info.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        "rows": [ "D4", "C6", "D7", "E6" ]
+    },
+    "diode_direction": "COL2ROW",
+    "split": {
+        "enabled": true,
+        "soft_serial_pin": "D2"
+    },
+    "layout_aliases": {
+        "LAYOUT": "LAYOUT_split_3x5_3"
+    },
```
Comment (zvecr) [inline keyboards/idank/spankbd/info.json:31]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/idank/spankbd/keymaps/via/config.h
[the remark this answers is not in the document]
Comment (idank) [reply keyboards/idank/spankbd/keymaps/via/config.h:1]: Done.

File: keyboards/idank/spankbd/keymaps/default/config.h
Comment (idank) [reply keyboards/idank/spankbd/keymaps/default/config.h:1]: Done.

File: keyboards/idank/spankbd/info.json (hunk 1 above)
Comment (idank) [reply keyboards/idank/spankbd/info.json]: Done.

File: keyboards/idank/spankbd/keymaps/default/keymap.json [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+          "KC_RBRC",
+          "KC_BTN1",
+          "KC_AMPR",
+          "KC_EQL",
+          "KC_COMM",
+          "KC_DOT",
+          "KC_MINS",
+          "KC_BSPC",
+          "KC_SCLN",
+          "KC_EQL",
```
Comment (waffle87) [inline keyboards/idank/spankbd/keymaps/default/keymap.json:80]:
```suggestion
          "_______",
```

File: keyboards/idank/spankbd/info.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        "pid": "0x3140",
+        "device_version": "0.0.1"
+    },
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (waffle87) [inline keyboards/idank/spankbd/info.json:18]:
```suggestion
        "nkro": true,
        "oled": false
```

File: keyboards/idank/spankbd/rules.mk [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# OLED disabled by default, enable with OLED_ENABLE = yes.
+OLED_ENABLE = no
+OLED_DRIVER = ssd1306
```
Comment (waffle87) [inline keyboards/idank/spankbd/rules.mk:3]:
```suggestion
# This file intentionally left blank
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/idank/spankbd/info.json (hunk 3 above)
Comment (idank) [reply keyboards/idank/spankbd/info.json]: done, thanks

File: keyboards/idank/spankbd/rules.mk (hunk 4 above)
Comment (idank) [reply keyboards/idank/spankbd/rules.mk]: done, thanks

File: keyboards/idank/spankbd/keymaps/default/keymap.json (hunk 2 above)
Comment (idank) [reply keyboards/idank/spankbd/keymaps/default/keymap.json:80]: was this intentional?

File: keyboards/idank/spankbd/info.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        "device_version": "0.0.1"
+    },
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "oled": false
```
Comment (drashna) [inline keyboards/idank/spankbd/info.json:19]:
```suggestion
        "extrakey": true,
        "mousekey": true,
        "nkro": true
```

File: keyboards/idank/spankbd/keymaps/default/keymap.json (hunk 2 above)
Comment (drashna) [reply keyboards/idank/spankbd/keymaps/default/keymap.json:80]: > was this intentional?

A better question is, how would you hit this key normally, as you'd have to be holding down `LT(1, KC_SPC)` to get to this layer?  And with a lack of any other layer keycodes to get to this layer ... what's the point of having a keycode here, when the transparent key is a good placeholder to note that this spot is taken/unusable?  

Which is most likely the motivation behind the request.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/idank/spankbd/keymaps/default/keymap.json (hunk 2 above)
Comment (idank) [reply keyboards/idank/spankbd/keymaps/default/keymap.json]: done

File: keyboards/idank/spankbd/info.json (hunk 5 above)
Comment (idank) [reply keyboards/idank/spankbd/info.json]: done, thanks

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**